File: 58A-Chat room/chatroom.cpp
#include<iostream>

using namespace std;

int main () {
    string word;
    cin >> word;
    string word2 = "hello";
    int j = 0;
    int pas = 0;
    for(int i = 0; i < word.length(); i++){
        if(word[i] == word2[j]){
            j++;
            pas++;
        }
        if(pas == 5){
            break;
        }
    }
    if (pas == 5){
        cout << "YES";
    }
    else{
        cout << "NO";
    }
}

File: 58A-Chat room/chatroom.py
def main():
    word = input()
    word2 = "hello"
    pas = 0
    j = 0
    for i in range (0,len(word)):
        if word[i] == word2[j]:
            j += 1
            pas += 1

        if pas == 5:
            break

    if pas == 5:
        print("YES")
        
    else:
        print("NO")


main()